drivers/net/bnxt/bnxt_hwrm.c

```diff
@@ -1603,14 +1629,28 @@ int bnxt_hwrm_allocate_vfs(struct bnxt *bp, int num_vfs)
 		rc = bnxt_hwrm_send_message(bp, &req, sizeof(req));
 		HWRM_CHECK_RESULT;
 
-		bp->pf.max_rsscos_ctx -= rte_le_to_cpu_16(req.num_rsscos_ctxs);
-		bp->pf.max_stat_ctx -= rte_le_to_cpu_16(req.num_stat_ctxs);
-		bp->pf.max_cp_rings -= rte_le_to_cpu_16(req.num_cmpl_rings);
-		bp->pf.max_tx_rings -= rte_le_to_cpu_16(req.num_tx_rings);
-		bp->pf.max_rx_rings -= rte_le_to_cpu_16(req.num_rx_rings);
-		bp->pf.max_l2_ctx -= rte_le_to_cpu_16(req.num_l2_ctxs);
-		bp->pf.max_vnics -= rte_le_to_cpu_16(req.num_vnics);
-		bp->max_ring_grps -= rte_le_to_cpu_16(req.num_hw_ring_grps);
+		/* Get the actual allocated values now */
+		HWRM_PREP(qcreq, FUNC_QCAPS, -1, qcresp);
+		qcreq.fid = rte_cpu_to_le_16(bp->pf.first_vf_id + i);
+		rc = bnxt_hwrm_send_message(bp, &qcreq, sizeof(req));
+		if (rc) {
+			RTE_LOG(ERR, PMD, "hwrm_func_qcaps failed rc:%d\n", rc);
+			copy_func_cfg_to_qcaps(&req, qcresp);
+		}
+		else if (resp->error_code) {
+			rc = rte_le_to_cpu_16(resp->error_code);
+			RTE_LOG(ERR, PMD, "hwrm_finc_qcaps error %d\n", rc);
+			copy_func_cfg_to_qcaps(&req, qcresp);
+		}
+
```

Comment:
If we are doing the copy_func_cfg_to_qcaps() should we still return rc and roll back in bnxt_ethdev.c ?